lib/ProductOpener/Ecoscore.pm

```diff
@@ -188,51 +278,85 @@ sub compute_ecoscore($) {
+		$product_ref->{ecoscore_data}{agribalyse}{agribalyse_food_name_en} = $agribalyse{$agb}{name_en};
+		$product_ref->{ecoscore_data}{agribalyse}{agribalyse_ef_total} = $agribalyse{$agb}{ef_total};
 		
 		# Formula to transform the Environmental Footprint single score to a 0 to 100 scale
 		# Note: EF score are for mPt / kg in Agribalyse, we need it in micro points per 100g
-		$ecoscore_score = -15 * log($agribalyse{$agb}{ef_total} * $agribalyse{$agb}{ef_total} * (1000 * 1000 / 100) + 195 ) + 178;
-	}
-	else {
-		# We need an AgriBalyse category match to compute the Eco-Score
-		$product_ref->{ecoscore_data}{status} = "unknown";
+		$product_ref->{ecoscore_data}{agribalyse}{score} = -15 * log($agribalyse{$agb}{ef_total} * $agribalyse{$agb}{ef_total} * (1000 * 1000 / 100) + 195 ) + 178;
 	}
+}
+
+
+=head2 compute_ecoscore_production_system_adjustment ( $product_ref )
+
+Computes an adjustment (bonus or malus) based on production system of the product (e.g. organic).
+
+=head3 Arguments
+
+=head4 Product reference $product_ref
+
+=head3 Return values
+
+The adjustment value and computations details are stored in the product reference passed as input parameter.
+
+Returned values:
+
+$product_ref->{adjustments}{production_system} hash with:
+- 
+
+$product_ref->{ecoscore_data}{missing} hash with:
+
+
+=cut
+
+my @production_system_labels = (
+	["fr:nature-et-progres", 20],
+	["fr:bio-coherence", 20],
+	["en:demeter", 20],
 	
-	if ($ecoscore_score) {
-		$product_ref->{ecoscore_data}{status} = "known";
-		$product_ref->{ecoscore_score} = $ecoscore_score;
+	["fr:ab-agriculture-biologique", 15],
+	["en:eu-organic", 15],
+	
+	["fr:haute-valeur-environnementale", 10],
+	["en:utz-certified", 10],
+	["en:rainforest-alliance", 10],
+	["en:fairtrade-international", 10],
+	["fr:bleu-blanc-coeur", 10],
+	["fr:label-rouge", 10],
+	["en:sustainable-seafood-msc", 10],
+	["en:responsible-aquaculture-asc", 10],
+);
+
+sub compute_ecoscore_production_system_adjustment($) {
+
+	my $product_ref = shift;
+	
+	$product_ref->{ecoscore_data}{adjustments}{production_system} = {};
 		
-		if ($product_ref->{ecoscore_score} >= 80) {
-			$product_ref->{ecoscore_grade} = "a";
-		}
-		elsif ($product_ref->{ecoscore_score} >= 60) {
-			$product_ref->{ecoscore_grade} = "b";
-		}
-		elsif ($product_ref->{ecoscore_score} >= 40) {
-			$product_ref->{ecoscore_grade} = "c";
-		}
-		elsif ($product_ref->{ecoscore_score} >= 20) {
-			$product_ref->{ecoscore_grade} = "d";
-		}
-		else {
-			$product_ref->{ecoscore_grade} = "e";
+	foreach my $label_ref (@production_system_labels) {
+		
+		my ($label, $value) = @$label_ref;
+		
+		if (has_tag($product_ref, "labels", $label)
+			# Label Rouge labels is counted only for beef, veal and lamb
+			and (($label ne "fr:label-rouge")
+				or (has_tag($product_ref, "categories", "en:beef"))
+				or (has_tag($product_ref, "categories", "en:veal-meat"))
+				or (has_tag($product_ref, "categories", "en:lamb-meat")))) {
```

Comment:
i don't know if other labels/categories are referred to in code, but is it worth putting a "do not rename from x" comment on the taxonomy entries, and/or unit tests to make sure they're not renamed?

Reply:
It's a good idea. Another way could be to canonicalize the tags once during the module initialization, so that it still works if we switch the canonical name with a synonym.

Reply:
ok I'm going to add some tests, but they are going to fail until https://github.com/openfoodfacts/openfoodfacts-server/pull/4385 is merged (because I renamed a couple of labels from en: to fr: )

so I will first merge, and then make a new PR for the tests